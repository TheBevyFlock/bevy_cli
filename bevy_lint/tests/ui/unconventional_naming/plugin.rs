#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::unconventional_naming)]
//~^ NOTE: the lint level is defined here
#![allow(dead_code)]
use bevy::prelude::*;

// This should raise an error, since it does not end in "Plugin".
struct Foo;
//~^ ERROR: unconventional type name for a `Plugin`
//~| NOTE: structures that implement `Plugin` should end in `Plugin`
//~| HELP: rename `Foo`

//~v NOTE: `Plugin` implemented here
impl Plugin for Foo {
    fn build(&self, _app: &mut App) {}
}

// This should _not_ raise an error, since it ends in "Plugin".
struct BarPlugin;

impl Plugin for BarPlugin {
    fn build(&self, _app: &mut App) {}
}

// Though this does not end in "Plugin", the lint is silenced, so no error is raised.
#[allow(bevy::unconventional_naming)]
struct Baz;

impl Plugin for Baz {
    fn build(&self, _app: &mut App) {}
}

fn main() {
    App::new().add_plugins((Foo, BarPlugin, Baz));
}
