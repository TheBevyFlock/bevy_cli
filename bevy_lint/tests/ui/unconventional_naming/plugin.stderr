error: unconventional type name for a `Plugin`
  --> tests/ui/unconventional_naming/plugin.rs:9:8
   |
9  | struct Foo;
   |        ^^^ help: rename `Foo`: `FooPlugin`
   |
note: structures that implement `Plugin` should end in `Plugin`
  --> tests/ui/unconventional_naming/plugin.rs:9:8
   |
9  | struct Foo;
   |        ^^^
note: `Plugin` implemented here
  --> tests/ui/unconventional_naming/plugin.rs:15:1
   |
15 | / impl Plugin for Foo {
16 | |     fn build(&self, _app: &mut App) {}
17 | | }
   | |_^
note: the lint level is defined here
  --> tests/ui/unconventional_naming/plugin.rs:3:9
   |
3  | #![deny(bevy::unconventional_naming)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

